pkg/ir/spec.go

```diff
@@ -75,20 +84,31 @@ type (
 	Definitions struct {
 		ConfigDefs
 
+		BlockedResources
```

Comment:
I think `BlockedResources` fit better inside the `Spec` struct, as they do not come from the config file directly, but are rather the result of combining the config and the required-connection spec.
Please see if moving them is easy.

Reply:
done